packages/hardhat-cannon/src/tasks/verify.ts

```diff
@@ -50,18 +51,20 @@ task(TASK_VERIFY, 'Verify a package on Etherscan')
     }
 
     for (const c in outputs.contracts) {
-      console.log('Verifying contract:', c);
-      try {
-        await hre.run('verify:verify', {
-          contract: `${outputs.contracts[c].sourceName}:${outputs.contracts[c].contractName}`,
-          address: outputs.contracts[c].address,
-          constructorArguments: outputs.contracts[c].constructorArgs || [],
-        });
-      } catch (err) {
-        if ((err as Error).message.includes('Already Verified')) {
-          console.log('Already verified');
-        } else {
-          throw err;
+      if (_.get(outputs, ['contracts', c, 'sourceName']) && _.get(outputs, _.get(outputs, ['contracts', c, 'contractName'])) {
+        console.log('Verifying contract:', c);
+        try {
+          await hre.run('verify:verify', {
+            contract: `${outputs.contracts[c].sourceName}:${outputs.contracts[c].contractName}`,
+            address: outputs.contracts[c].address,
+            constructorArguments: outputs.contracts[c].constructorArgs || [],
+          });
+        } catch (err) {
+          console.log(
+            `Unable to verify ${outputs.contracts[c].sourceName}:${outputs.contracts[c].contractName} - ${
```

Comment:
ordinarily I would suggest doing the error parsing like we were doing before but as we saw the upstream error could change or be different depending on the release so probably just failing gracefully here for now is indeed the best overall path.